crates/ruff_python_semantic/src/model.rs

```diff
@@ -662,6 +662,91 @@ impl<'a> SemanticModel<'a> {
                 }
             }
 
+            // FIXME: Shouldn't this happen above where `class_variables_visible` is set?
+            seen_function |= scope.kind.is_function();
```

Comment:
Can you expand on why you think this should happen above? 

My assumption is that it's here because it updates the state for the next iteration. The only possible issue I see is that it fails updating the `seen_function` if the `self.bindings[binding_id].kind` is an `Annotation`. I, unfortunately, don't have a good enough understanding of this code to assess whether this is a) possible and b) a problem

Reply:
I'm fairly confident it's incorrect to skip updating `seen_function` if you see a `BindingKind::Annotation`.

That being said, situations where this would actually lead to the wrong results should be incredibly rare, since the corresponding Python code itself would be highly suspect and probably incorrect.